nifty-core/src/main/java/com/facebook/nifty/ssl/SslHandlerFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2012-2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import org.jboss.netty.handler.ssl.SslHandler;
+
+/**
+ * A Factory that returns a handler.
+ */
+public interface SslHandlerFactory {
```

Comment:
ya I thought about that, but that mixes with dependency inject to inject the instance. We might want to have the securityFactory and the sslhandlerfactory together for certain servers to support both security protocols. We could technically bind them with annotations, however I haven't seen a use case where SslHandlerFactory needs to be bound via DI.